File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/SimplePasswordPreference.java
package org.totschnig.myexpenses.preference;

import android.content.Context;
import android.util.AttributeSet;

import org.totschnig.myexpenses.R;

import androidx.annotation.Nullable;
import androidx.preference.DialogPreference;

public class SimplePasswordPreference extends DialogPreference {
  public SimplePasswordPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
    super(context, attrs, defStyleAttr, defStyleRes);
    init();
  }
  public SimplePasswordPreference(Context context, AttributeSet attrs, int defStyleAttr) {
    super(context, attrs, defStyleAttr);
    init();
  }

  public SimplePasswordPreference(Context context, AttributeSet attrs) {
    super(context, attrs);
    init();
  }

  public SimplePasswordPreference(Context context) {
    super(context);
    init();
  }

  private void init() {
    setDialogLayoutResource(R.layout.simple_password_dialog);
  }

  @Nullable
  public String getValue() {
    return getPersistedString(null);
  }

  public void setValue(String value) {
    persistString(value);
  }
}

File: myExpenses/src/main/java/org/totschnig/myexpenses/preference/SimplePasswordPreference.kt
package org.totschnig.myexpenses.preference

import android.content.Context
import android.util.AttributeSet
import androidx.preference.DialogPreference
import org.totschnig.myexpenses.R

class SimplePasswordPreference @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null
) : DialogPreference(context, attrs) {

    private fun init() {
        dialogLayoutResource = R.layout.simple_password_dialog
    }

    var value: String?
        get() = getPersistedString(null)
        set(value) {
            persistString(value)
        }

    init {
        init()
    }
}